dashboard/src/apollo/schema/combinedTypes.macro.js

```diff
@@ -0,0 +1,39 @@
+/* eslint-disable import/no-webpack-loader-syntax, import/extensions */
+
+import { buildSchema, execute, parse } from "graphql";
+import rawSchema from "!!raw-loader!./combined.graphql";
+
+export default ({ emitFile }) => {
+  //
+  // Apollo only needs to be aware of the possible types unions and interfaces
+  // may contain. So instead of retrieving the entire schema, we simply retrieve
+  // the names and possibleTypes.
+  //
+  // More: https://www.apollographql.com/docs/react/advanced/fragments.html#fragment-matcher
+  //
+  const schema = buildSchema(rawSchema);
+  const queryAST = parse(
+    `
+      query IntrospectionQuery {
+        __schema {
+          types {
+            kind
+            name
+            possibleTypes {
+              name
+            }
+          }
+        }
+      }
+    `,
+  );
+  const result = execute(schema, queryAST);
+  if (result.errors) throw result.errors;
+
+  // https://github.com/apollographql/apollo-client/blob/2701b0acb89711864bde28341cb5cfcf909d2caf/packages/apollo-cache-inmemory/src/fragmentMatcher.ts#L149-L155
+  const filteredTypes = result.data.__schema.types.filter(
+    type => type.kind === "UNION" || type.kind === "INTERFACE",
+  );
+  result.data.__schema.types = filteredTypes;
+  return emitFile("combinedTypes.json", JSON.stringify(result));
```

Comment:
You don't need to call `emitFile` at all here. You can simply return the module contents.